pkg/health/condition/check_backup_ready.go

```diff
@@ -48,6 +49,34 @@ func (a *backupReadyCheck) Check(ctx context.Context, etcd druidv1alpha1.Etcd) R
 		message: "Cannot determine etcd backup status",
 	}
 
+	// Special case of etcd not being configured to take snapshots
+	// Do not add the BackupReady condition if backup is not configured
+	if etcd.Spec.Backup.Store == nil {
+		return nil
```

Comment:
Can you double check if this also works when backups are disabled afterwards, i.e. it was enabled before? I assume that the Backup condition will still be present and we probably want to condition to be removed. WDYT?

Reply:
Hmm, I checked this and it does not remove the condition in this case
Let me try and see if we can support this case

Reply:
Updated the builder function to not add any old conditions to the condition set
PTAL whenever possible

Reply:
To be on the safe side, can you only not add old conditions for known types `ConditionTypeReady`, `ConditionTypeAllMembersReady` and `ConditionTypeBackupReady`?

Reply:
Updated to include this